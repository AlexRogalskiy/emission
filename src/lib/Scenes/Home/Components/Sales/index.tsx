import { Sales_query } from "__generated__/Sales_query.graphql"
import React from "react"
import { RefreshControl, SectionList, StyleSheet } from "react-native"
import { createRefetchContainer, graphql, RelayRefetchProp } from "react-relay"
import LotsByFollowedArtists from "./Components/LotsByFollowedArtists"
import { SaleList } from "./Components/SaleList"
import { ZeroState } from "./Components/ZeroState"

interface Props {
  relay: RelayRefetchProp
  query: Sales_query
}

interface State {
  isRefreshing: boolean
}

class Sales extends React.Component<Props, State> {
  state = {
    isRefreshing: false,
  }

  get data() {
    const sales = this.props.query.salesConnection.edges.map(({ node }) => node)
    const liveAuctions = sales.filter(a => !!a.live_start_at)
    const timedAuctions = sales.filter(a => !a.live_start_at)

    return {
      liveAuctions,
      timedAuctions,
    }
  }

  handleRefresh = () => {
    this.setState({ isRefreshing: true })
    this.props.relay.refetch(
      {},
      {},
      error => {
        if (error) {
          // FIXME: Handle error
          console.error("Sales/index.tsx", error.message)
        }
        this.setState({ isRefreshing: false })
      },
      { force: true }
    )
  }

  render() {
    if (this.props.query.salesConnection.edges.length === 0) {
      return <ZeroState />
    }

    const sections = [
      {
        data: [{ data: this.data.liveAuctions }],
        title: "Current Live Auctions",
        isFirstItem: true,
        renderItem: props => <SaleList {...props} />,
      },
      {
        data: [{ data: this.data.timedAuctions }],
        title: "Current Timed Auctions",
        renderItem: props => <SaleList {...props} />,
      },
      {
        data: [{ data: this.props.query }],
        title: "Lots by Artists You Follow",
        renderItem: props => {
          return <LotsByFollowedArtists title={props.section.title} query={props.item.data} />
        },
      },
    ]

    return (
      <SectionList
        contentContainerStyle={SectionListStyles.contentContainer}
        stickySectionHeadersEnabled={false}
        sections={sections}
        keyExtractor={item => item.id}
        renderItem={() => undefined}
        refreshControl={<RefreshControl refreshing={this.state.isRefreshing} onRefresh={this.handleRefresh} />}
      />
    )
  }
}

export default createRefetchContainer(
  Sales,
  {
    query: graphql`
      fragment Sales_query on Query {
        salesConnection(live: true, isAuction: true, first: 100, sort: TIMELY_AT_NAME_ASC) {
          edges {
            node {
              ...SaleListItem_sale
              live_start_at: liveStartAt
            }
          }
        }
        ...LotsByFollowedArtists_query
      }
    `,
  },
  graphql`
    query SalesQuery {
      ...Sales_query
    }
  `
)

const SectionListStyles = StyleSheet.create({
  contentContainer: {
    justifyContent: "space-between",
    paddingTop: 2,
    padding: 10,
    display: "flex",
  },
})
