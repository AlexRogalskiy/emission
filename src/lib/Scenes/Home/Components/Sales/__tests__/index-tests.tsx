import { shallow } from "enzyme"
import React from "react"
import "react-native"

import { renderWithLayout } from "lib/tests/renderWithLayout"

import Sales from "../index"

jest.mock("WebView", () => "WebView")
jest.mock("../Components/ZeroState/index.html", () => "")
jest.mock("../Components/LotsByFollowedArtists", () => "")

it("renders the ZeroState when there are no sales", () => {
  const auctions = shallow(<Sales {...props} query={{ salesConnection: { edges: [] } } as any} />)
  expect(auctions.find("ZeroState").length).toEqual(1)
})

it("looks correct when rendered", () => {
  const auctions = renderWithLayout(<Sales {...props as any} />, { width: 1000 })
  expect(auctions).toMatchSnapshot()
})

const props = {
  relay: {
    environment: null,
    hasMore: jest.fn(),
    isLoading: jest.fn(),
    loadMore: jest.fn(),
    refetch: jest.fn(),
  },
  query: {
    salesConnection: {
      edges: [
        {
          node: {
            id: "wright-noma",
            href: "/auction/wright-noma",
            name: "Wright: noma",
            is_open: true,
            is_live_open: true,
            start_at: "2017-10-16T22:00:00+00:00",
            end_at: null,
            registration_ends_at: "2017-11-01T13:00:00+00:00",
            live_start_at: "2017-11-02T13:00:00+00:00",
            displayTimelyAt: "In Progress",
            cover_image: {
              url: "https://d32dm0rphc51dk.cloudfront.net/WV-7BYlETayN8MGkNjOGXw/source.jpg",
            },
          },
        },
        {
          node: {
            id: "freemans-modern-and-contemporary-works-of-art",
            href: "/auction/freemans-modern-and-contemporary-works-of-art",
            name: "Freeman's: Modern & Contemporary Works of Art",
            is_open: true,
            is_live_open: false,
            start_at: "2017-10-17T15:00:00+00:00",
            end_at: null,
            registration_ends_at: "2017-11-01T17:00:00+00:00",
            live_start_at: "2017-11-02T17:00:00+00:00",
            displayTimelyAt: "In Progress",
            cover_image: {
              url: "https://d32dm0rphc51dk.cloudfront.net/eeqLfwMMAYA8XOmeYEb7Rg/source.jpg",
            },
          },
        },
      ],
    },
    me: {
      lotsByFollowedArtists: {
        pageInfo: {},
        edges: [
          {
            node: {
              id: "foo",
            },
          },
        ],
      },
    },
  },
}
