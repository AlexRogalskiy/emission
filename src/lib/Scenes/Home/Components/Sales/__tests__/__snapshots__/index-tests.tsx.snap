// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`looks correct when rendered 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "display": "flex",
      "justifyContent": "space-between",
      "padding": 10,
      "paddingTop": 2,
    }
  }
  data={
    Array [
      Object {
        "data": Array [
          Object {
            "data": Array [
              Object {
                "cover_image": Object {
                  "url": "https://d32dm0rphc51dk.cloudfront.net/WV-7BYlETayN8MGkNjOGXw/source.jpg",
                },
                "displayTimelyAt": "In Progress",
                "end_at": null,
                "href": "/auction/wright-noma",
                "id": "wright-noma",
                "is_live_open": true,
                "is_open": true,
                "live_start_at": "2017-11-02T13:00:00+00:00",
                "name": "Wright: noma",
                "registration_ends_at": "2017-11-01T13:00:00+00:00",
                "start_at": "2017-10-16T22:00:00+00:00",
              },
              Object {
                "cover_image": Object {
                  "url": "https://d32dm0rphc51dk.cloudfront.net/eeqLfwMMAYA8XOmeYEb7Rg/source.jpg",
                },
                "displayTimelyAt": "In Progress",
                "end_at": null,
                "href": "/auction/freemans-modern-and-contemporary-works-of-art",
                "id": "freemans-modern-and-contemporary-works-of-art",
                "is_live_open": false,
                "is_open": true,
                "live_start_at": "2017-11-02T17:00:00+00:00",
                "name": "Freeman's: Modern & Contemporary Works of Art",
                "registration_ends_at": "2017-11-01T17:00:00+00:00",
                "start_at": "2017-10-17T15:00:00+00:00",
              },
            ],
          },
        ],
        "isFirstItem": true,
        "renderItem": [Function],
        "title": "Current Live Auctions",
      },
      Object {
        "data": Array [
          Object {
            "data": Array [],
          },
        ],
        "renderItem": [Function],
        "title": "Current Timed Auctions",
      },
      Object {
        "data": Array [
          Object {
            "data": Object {
              "me": Object {
                "lotsByFollowedArtists": Object {
                  "edges": Array [
                    Object {
                      "node": Object {
                        "id": "foo",
                      },
                    },
                  ],
                  "pageInfo": Object {},
                },
              },
              "salesConnection": Object {
                "edges": Array [
                  Object {
                    "node": Object {
                      "cover_image": Object {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/WV-7BYlETayN8MGkNjOGXw/source.jpg",
                      },
                      "displayTimelyAt": "In Progress",
                      "end_at": null,
                      "href": "/auction/wright-noma",
                      "id": "wright-noma",
                      "is_live_open": true,
                      "is_open": true,
                      "live_start_at": "2017-11-02T13:00:00+00:00",
                      "name": "Wright: noma",
                      "registration_ends_at": "2017-11-01T13:00:00+00:00",
                      "start_at": "2017-10-16T22:00:00+00:00",
                    },
                  },
                  Object {
                    "node": Object {
                      "cover_image": Object {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/eeqLfwMMAYA8XOmeYEb7Rg/source.jpg",
                      },
                      "displayTimelyAt": "In Progress",
                      "end_at": null,
                      "href": "/auction/freemans-modern-and-contemporary-works-of-art",
                      "id": "freemans-modern-and-contemporary-works-of-art",
                      "is_live_open": false,
                      "is_open": true,
                      "live_start_at": "2017-11-02T17:00:00+00:00",
                      "name": "Freeman's: Modern & Contemporary Works of Art",
                      "registration_ends_at": "2017-11-01T17:00:00+00:00",
                      "start_at": "2017-10-17T15:00:00+00:00",
                    },
                  },
                ],
              },
            },
          },
        ],
        "renderItem": [Function],
        "title": "Lots by Artists You Follow",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  renderItem={[Function]}
  scrollEventThrottle={50}
  sections={
    Array [
      Object {
        "data": Array [
          Object {
            "data": Array [
              Object {
                "cover_image": Object {
                  "url": "https://d32dm0rphc51dk.cloudfront.net/WV-7BYlETayN8MGkNjOGXw/source.jpg",
                },
                "displayTimelyAt": "In Progress",
                "end_at": null,
                "href": "/auction/wright-noma",
                "id": "wright-noma",
                "is_live_open": true,
                "is_open": true,
                "live_start_at": "2017-11-02T13:00:00+00:00",
                "name": "Wright: noma",
                "registration_ends_at": "2017-11-01T13:00:00+00:00",
                "start_at": "2017-10-16T22:00:00+00:00",
              },
              Object {
                "cover_image": Object {
                  "url": "https://d32dm0rphc51dk.cloudfront.net/eeqLfwMMAYA8XOmeYEb7Rg/source.jpg",
                },
                "displayTimelyAt": "In Progress",
                "end_at": null,
                "href": "/auction/freemans-modern-and-contemporary-works-of-art",
                "id": "freemans-modern-and-contemporary-works-of-art",
                "is_live_open": false,
                "is_open": true,
                "live_start_at": "2017-11-02T17:00:00+00:00",
                "name": "Freeman's: Modern & Contemporary Works of Art",
                "registration_ends_at": "2017-11-01T17:00:00+00:00",
                "start_at": "2017-10-17T15:00:00+00:00",
              },
            ],
          },
        ],
        "isFirstItem": true,
        "renderItem": [Function],
        "title": "Current Live Auctions",
      },
      Object {
        "data": Array [
          Object {
            "data": Array [],
          },
        ],
        "renderItem": [Function],
        "title": "Current Timed Auctions",
      },
      Object {
        "data": Array [
          Object {
            "data": Object {
              "me": Object {
                "lotsByFollowedArtists": Object {
                  "edges": Array [
                    Object {
                      "node": Object {
                        "id": "foo",
                      },
                    },
                  ],
                  "pageInfo": Object {},
                },
              },
              "salesConnection": Object {
                "edges": Array [
                  Object {
                    "node": Object {
                      "cover_image": Object {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/WV-7BYlETayN8MGkNjOGXw/source.jpg",
                      },
                      "displayTimelyAt": "In Progress",
                      "end_at": null,
                      "href": "/auction/wright-noma",
                      "id": "wright-noma",
                      "is_live_open": true,
                      "is_open": true,
                      "live_start_at": "2017-11-02T13:00:00+00:00",
                      "name": "Wright: noma",
                      "registration_ends_at": "2017-11-01T13:00:00+00:00",
                      "start_at": "2017-10-16T22:00:00+00:00",
                    },
                  },
                  Object {
                    "node": Object {
                      "cover_image": Object {
                        "url": "https://d32dm0rphc51dk.cloudfront.net/eeqLfwMMAYA8XOmeYEb7Rg/source.jpg",
                      },
                      "displayTimelyAt": "In Progress",
                      "end_at": null,
                      "href": "/auction/freemans-modern-and-contemporary-works-of-art",
                      "id": "freemans-modern-and-contemporary-works-of-art",
                      "is_live_open": false,
                      "is_open": true,
                      "live_start_at": "2017-11-02T17:00:00+00:00",
                      "name": "Freeman's: Modern & Contemporary Works of Art",
                      "registration_ends_at": "2017-11-01T17:00:00+00:00",
                      "start_at": "2017-10-17T15:00:00+00:00",
                    },
                  },
                ],
              },
            },
          },
        ],
        "renderItem": [Function],
        "title": "Lots by Artists You Follow",
      },
    ]
  }
  stickyHeaderIndices={Array []}
  stickySectionHeadersEnabled={false}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <RCTRefreshControl />
  <View>
    <View
      onLayout={[Function]}
      style={null}
    />
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "marginTop": -9,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "marginBottom": -4,
                "paddingBottom": 0,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 25,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "ReactNativeAGaramondPro-Regular",
                  "fontSize": 30,
                  "textAlign": "left",
                },
              ]
            }
          >
            Current Live Auctions
          </Text>
        </View>
        <View
          contentContainerStyle={
            Object {
              "display": "flex",
              "justifyContent": "space-between",
              "padding": 5,
            }
          }
          data={
            Array [
              Object {
                "cover_image": Object {
                  "url": "https://d32dm0rphc51dk.cloudfront.net/WV-7BYlETayN8MGkNjOGXw/source.jpg",
                },
                "displayTimelyAt": "In Progress",
                "end_at": null,
                "href": "/auction/wright-noma",
                "id": "wright-noma",
                "is_live_open": true,
                "is_open": true,
                "live_start_at": "2017-11-02T13:00:00+00:00",
                "name": "Wright: noma",
                "registration_ends_at": "2017-11-01T13:00:00+00:00",
                "start_at": "2017-10-16T22:00:00+00:00",
              },
              Object {
                "cover_image": Object {
                  "url": "https://d32dm0rphc51dk.cloudfront.net/eeqLfwMMAYA8XOmeYEb7Rg/source.jpg",
                },
                "displayTimelyAt": "In Progress",
                "end_at": null,
                "href": "/auction/freemans-modern-and-contemporary-works-of-art",
                "id": "freemans-modern-and-contemporary-works-of-art",
                "is_live_open": false,
                "is_open": true,
                "live_start_at": "2017-11-02T17:00:00+00:00",
                "name": "Freeman's: Modern & Contemporary Works of Art",
                "registration_ends_at": "2017-11-01T17:00:00+00:00",
                "start_at": "2017-10-17T15:00:00+00:00",
              },
            ]
          }
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={4}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "height": 0,
                      "marginBottom": 10,
                      "marginLeft": 10,
                      "marginRight": 10,
                      "marginTop": 10,
                      "width": 0,
                    },
                  ]
                }
              >
                <AROpaqueImageView
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "left": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      },
                    ]
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "rgba(0, 0, 0, 0.5)",
                        "height": "100%",
                        "left": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "flex-start",
                          "display": "flex",
                          "flexDirection": "row",
                          "flexWrap": "nowrap",
                          "justifyContent": "space-between",
                          "paddingBottom": 10,
                          "paddingLeft": 10,
                          "paddingRight": 10,
                          "paddingTop": 10,
                          "width": "100%",
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={2}
                      style={
                        Array [
                          Object {
                            "fontSize": 17,
                          },
                          Array [
                            Object {
                              "color": "white",
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "flexShrink": 1,
                              "fontSize": 15,
                            },
                          ],
                          Object {
                            "fontFamily": "AGaramondPro-Regular",
                          },
                        ]
                      }
                    >
                      Wright: noma
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "bottom": 0,
                          "marginBottom": 5,
                          "marginLeft": 5,
                          "marginRight": 5,
                          "marginTop": 5,
                          "paddingBottom": 5,
                          "paddingLeft": 5,
                          "paddingRight": 5,
                          "paddingTop": 5,
                          "position": "absolute",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontFamily": "AvantGardeGothicITC",
                            "fontSize": 10,
                            "letterSpacing": 1.5,
                          },
                        ]
                      }
                    >
                      IN PROGRESS
                    </Text>
                  </View>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "height": 0,
                      "marginBottom": 10,
                      "marginLeft": 10,
                      "marginRight": 10,
                      "marginTop": 10,
                      "width": 0,
                    },
                  ]
                }
              >
                <AROpaqueImageView
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "left": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      },
                    ]
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "rgba(0, 0, 0, 0.5)",
                        "height": "100%",
                        "left": 0,
                        "position": "absolute",
                        "top": 0,
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "flex-start",
                          "display": "flex",
                          "flexDirection": "row",
                          "flexWrap": "nowrap",
                          "justifyContent": "space-between",
                          "paddingBottom": 10,
                          "paddingLeft": 10,
                          "paddingRight": 10,
                          "paddingTop": 10,
                          "width": "100%",
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={2}
                      style={
                        Array [
                          Object {
                            "fontSize": 17,
                          },
                          Array [
                            Object {
                              "color": "white",
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "flexShrink": 1,
                              "fontSize": 15,
                            },
                          ],
                          Object {
                            "fontFamily": "AGaramondPro-Regular",
                          },
                        ]
                      }
                    >
                      Freeman's: Modern & Contemporary Works of Art
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "bottom": 0,
                          "marginBottom": 5,
                          "marginLeft": 5,
                          "marginRight": 5,
                          "marginTop": 5,
                          "paddingBottom": 5,
                          "paddingLeft": 5,
                          "paddingRight": 5,
                          "paddingTop": 5,
                          "position": "absolute",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "white",
                            "fontFamily": "AvantGardeGothicITC",
                            "fontSize": 10,
                            "letterSpacing": 1.5,
                          },
                        ]
                      }
                    >
                      IN PROGRESS
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
    <View
      onLayout={[Function]}
      style={null}
    />
    <View
      onLayout={[Function]}
      style={null}
    />
    <View
      onLayout={[Function]}
      style={null}
    />
    <View
      onLayout={[Function]}
      style={null}
    />
    <View
      onLayout={[Function]}
      style={null}
    >
      <
        query={
          Object {
            "me": Object {
              "lotsByFollowedArtists": Object {
                "edges": Array [
                  Object {
                    "node": Object {
                      "id": "foo",
                    },
                  },
                ],
                "pageInfo": Object {},
              },
            },
            "salesConnection": Object {
              "edges": Array [
                Object {
                  "node": Object {
                    "cover_image": Object {
                      "url": "https://d32dm0rphc51dk.cloudfront.net/WV-7BYlETayN8MGkNjOGXw/source.jpg",
                    },
                    "displayTimelyAt": "In Progress",
                    "end_at": null,
                    "href": "/auction/wright-noma",
                    "id": "wright-noma",
                    "is_live_open": true,
                    "is_open": true,
                    "live_start_at": "2017-11-02T13:00:00+00:00",
                    "name": "Wright: noma",
                    "registration_ends_at": "2017-11-01T13:00:00+00:00",
                    "start_at": "2017-10-16T22:00:00+00:00",
                  },
                },
                Object {
                  "node": Object {
                    "cover_image": Object {
                      "url": "https://d32dm0rphc51dk.cloudfront.net/eeqLfwMMAYA8XOmeYEb7Rg/source.jpg",
                    },
                    "displayTimelyAt": "In Progress",
                    "end_at": null,
                    "href": "/auction/freemans-modern-and-contemporary-works-of-art",
                    "id": "freemans-modern-and-contemporary-works-of-art",
                    "is_live_open": false,
                    "is_open": true,
                    "live_start_at": "2017-11-02T17:00:00+00:00",
                    "name": "Freeman's: Modern & Contemporary Works of Art",
                    "registration_ends_at": "2017-11-01T17:00:00+00:00",
                    "start_at": "2017-10-17T15:00:00+00:00",
                  },
                },
              ],
            },
          }
        }
        title="Lots by Artists You Follow"
      />
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;
